lib/atom/atom-text-editor.js

```diff
@@ -77,6 +82,13 @@ export default class AtomTextEditor extends React.Component {
 
     this.refParent.map(element => {
       const editor = new TextEditor(modelProps);
+      editor.getElement().tabIndex = this.props.tabIndex;
+      if (this.props.className) {
+        editor.getElement().classList.add(this.props.className);
+      }
+      if (this.props.preselect) {
+        editor.selectAll();
```

Comment:
I went back and forth between supporting "all text preselected" and "cursor positioned at the end of the provided buffer." I think I landed on "all text preselected" because it's slightly more flexible (one keypress to decline a suggested starting value!) and because it's slightly easier to code (I don't have to pull anything from the buffer here.)